### DISCUSSION TITLE
Expose dialect internals for SQLc support

### DISCUSSION THREAD

**Discussion by casimcdaniels** [marcboeker/go-duckdb#316] [Ideas]
It would be awesome and an amazing data engineering workflow to be able to write SQL and generate go bindings to results using sqlc. Per [this discussion](https://github.com/sqlc-dev/sqlc/discussions/2515), there is interest (including me) but this package does not expose the parser and dialect internals to support adapting parsing logic to the generation engine.  What do you think about adding support for this so work can be started on integrating the dialect into sqlc?

**Comment by taniabogatsch** [comment]
Hi, @casimcdaniels - go-duckdb only has access to any functionality exposed by duckdb's C API.
It would be interesting to understand the specifics of what would be necessary for `sqlc` to work with go-duckdb.
I.e., what is missing in the C API, and what is missing in go-duckdb? From there, we could discuss whether that is possible in go-duckdb and the amount of work involved.

As a side note, we (duckdb) are also looking into a sane way to generate lightweight Go bindings from our C API. These bindings are interesting as they would enable users to write duckdb extensions in Go. Possibly, the same bindings could eventually be used with sqlc. But that project is still in its infant stage. 

Here are the links; we'll move these into the duckdb repo, once/if they mature enough. :D https://github.com/taniabogatsch/goofy-duck (example extension) https://github.com/taniabogatsch/duckdb-go-api (first experiments with bindings).

**Comment by mlafeldt** [comment]
👋 Thanks for starting this discussion!

This repository has moved to https://github.com/duckdb/duckdb-go

If this topic is still relevant, please continue the conversation at the new location. I'm closing this now as part of the migration.

Please see https://github.com/marcboeker/go-duckdb/issues/565 for more details.
